src/essence/Tools/Isochrone/IsochroneTool.js

```diff
@@ -0,0 +1,539 @@
+//New Tool Template
+//In the very least, each tool needs to be defined through require.js and return
+// an object with 'make' and 'destroy' functions
+import * as d3 from 'd3';
+import L_ from '../../Basics/Layers_/Layers_';
+import Map_ from '../../Basics/Map_/Map_';
+import $ from 'jquery';
+import CursorInfo from '../../Ancillary/CursorInfo';
+//import F_ from '../../Basics/Formulae_/Formulae_';
+//import turf from 'turf';
+
+import IsochroneManager from './IsochroneTool_Manager';
+
+import './IsochroneTool.css';
+const L = window.L; //shhh, eslint...
+
+/*
+Handles map events, sidebar management, drawing layers and markers.
+Individual isochrones, from data gathering to modeling to analysis,
+are handled in IsochroneTool_Manager and its imports.
+*/
+
+//Add the tool markup if you want to do it this way
+// prettier-ignore
+const markup = [
+    "<div id='isochroneTool'>",
+        "<div id='isochroneToolHeader'>",
+            "<span id='isochroneToolTitle'>Isochrone</span>",
+            "<span id='iscNew'>",
+                "New",
+                "<i class='mdi mdi-plus mdi-18px'></i>",
+            "</span>",
+        "</div>",
+        "<ul id='isochroneOptionsContainer'></ul>",
+    "</div>"
+].join('\n');
+
+//Legacy color mapping function
+function hueMap(val) {
+    const hueToChannel = h => {
+        h = (h + 1530) % 1530;
+        if(h < 255) return h;
+        if(h < 765) return 255;
+        if(h < 1020) return 255 - (h % 255);
+        return 0;
+    }
+    const hue = Math.floor(val * 1050);
+    const r = hueToChannel(hue + 510);
+    const g = hueToChannel(hue);
+    const b = hueToChannel(hue - 510);
+    return [r, g, b];
+}
+
+//https://leafletjs.com/reference-1.7.1.html#gridlayer
+L.IsochroneLayer = L.GridLayer.extend({
+    //Override to make layer accept Bounds in place of LatLngBounds
+    //  (cleaner, supports polar projections better)
+    _isValidTile(coords) {
+        const bounds = this.options.bounds;
+        return coords.x >= bounds.min.x
+            && coords.y >= bounds.min.y
+            && coords.x < bounds.max.x
+            && coords.y < bounds.max.y;
+    },
+    createTile: function(coords) {
+        const tile = L.DomUtil.create("canvas", "leaflet-tile");
+
+        const size = this.getTileSize();
+        tile.width = size.x;
+        tile.height = size.y;
+
+        const ctx = tile.getContext("2d");
+        const img = ctx.getImageData(0, 0, size.x, size.y);
+
+        const bounds = this.options.bounds;
+        const tXOffset = coords.x - bounds.min.x;
+        const tYOffset = coords.y - bounds.min.y;
+
+        const alpha = Math.floor(this.options.opacity * 255);
+
+        let di = 0; //img data index
+        for(let y = 0; y < size.y; y++) {
```

Comment:
nit: always include a space between `for` and `(`, eg. `for (let y = 0; ...)`. same goes for `if`, `switch`, `while`, etc